Release/5.1 (#24)
stanwood/Macchiato_iOS: README.md

[![iOS 8+](https://img.shields.io/badge/iOS-9+-EB7943.svg)]()
[![Build Status](https://travis-ci.org/stanwood/Macchiato_iOS.svg?branch=master)](https://travis-ci.org/stanwood/Macchiato_iOS)

> Mastering UITesting takes time. While making sure to catche all edge cases, system alerts, and any uncommon UI solution developed under the hood, `Macchiato` allows you to simplify the process. Removing the burdon of covering edge cases, animations, networking, and system alerts, simply manage your test cases in a `JSON` file locally or remotly. 

Reviewer: ```suggestion
Mastering UITesting takes time. While making sure to catch all edge cases, system alerts, and any uncommon UI solution developed under the hood, `Macchiato` allows you to simplify the process. Removing the burden of covering edge cases, animations, networking, and system alerts, simply manage your test cases in a `JSON` file locally or remotely. 
```

> This is a perfect solution for developers that wnat to integrate their UITesting with their own CI solution.

Reviewer: ```suggestion
This is a perfect solution for developers that want to integrate their UITesting with their own CI solution.
```

##### Overview

> The test navigation works by querying `XCUIElement` & `XCUIElementQuery` types. Check out the navigation types for a full list [here](https://github.com/stanwood/Macchiato_iOS/blob/master/Macchiato/Classes/NavigationType.swift). The UI Testing tool identifies each element by either an index, or a key, for example:

Reviewer: ```suggestion
The test navigation works by querying `XCUIElement` & `XCUIElementQuery` types. Check out the navigation types for a full list [here](https://github.com/stanwood/Macchiato_iOS/blob/master/Macchiato/Classes/NavigationType.swift). The UI Testing tool identifies each element by either an index or a key, for example:
```

[StanwoodCore Full Doc](https://stanwood.github.io/Stanwood_Core)

### Creating Your Test Cases

#### Overview

Macchiato works by querying element types from the views hierarchy and they can be accessed by calling a custom key or an index. For example, if we look at the image below from develop.apple.com, we can see how the elements are laid out.

![View Hierchy](assets/view.png)

> This is a great example where we have a top `UIView`, which can be identified with a key, and a `UICellectionView`, which cells can be identified with an index.

Reviewer: ```suggestion
This is a great example where we have a top `UIView`, which can be identified with a key, and a `UICellectionView`, whose cells can be identified with an index.
```

	}
	```

For the full action list, please check [here](https://github.com/stanwood/Macchiato_iOS/blob/master/Macchiato/Classes/Action.swift)

> For the full navigation types, please check [here](https://github.com/stanwood/Macchiato_iOS/blob/master/Macchiato/Classes/NavigationType.swift). The UI Testing tool identifies each element by either an index, or a key.

Reviewer: ```suggestion
For the full navigation types, please check [here](https://github.com/stanwood/Macchiato_iOS/blob/master/Macchiato/Classes/NavigationType.swift). The UI Testing tool identifies each element by either an index or a key.
```

>Note: Element identifiers will be listed in each project documentation  under **UI Testing Identifiers**

#### Handling System Alerts

> `Macchiato` supports system alerts. To monitor system alerts, simply add `.monitor` to any navigation handle.

Reviewer: ```suggestion
`Macchiato` supports system alerts. To monitor system alerts simply add `.monitor` to any navigation handle.
```
